cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/RunEnvironment.java

```diff
@@ -107,4 +108,36 @@ public ExecutionPath getExecutionPath() {
     public Map<String, SerializableSessionObject> getSerializableDataMap() {
         return serializableDataMap;
     }
+
+    public RunEnvironment cloneWithoutStacks() {
+        @SuppressWarnings("unchecked")
+        Map<String, ? extends Serializable> systemPropertiesCopy =
+                (Map<String, ? extends Serializable>) SerializationUtils.clone((Serializable) systemProperties);
+
+        RunEnvironment runEnvironmentWithoutStacks = new RunEnvironment(systemPropertiesCopy);
+
+        runEnvironmentWithoutStacks.contextStack = new ContextStack();
+        runEnvironmentWithoutStacks.parentFlowStack = new ParentFlowStack();
+
+        @SuppressWarnings("unchecked")
+        Map<String, Serializable> callArgumentsCopy =
+                (Map<String, Serializable>) SerializationUtils.clone((Serializable) callArguments);
+        runEnvironmentWithoutStacks.callArguments = callArgumentsCopy;
+
+        runEnvironmentWithoutStacks.executionPath =
+                (ExecutionPath) SerializationUtils.clone(executionPath);
+
+        @SuppressWarnings("unchecked")
+        Map<String, SerializableSessionObject> serializableDataMapCopy =
+                (Map<String, SerializableSessionObject>) SerializationUtils.clone((Serializable) serializableDataMap);
+        runEnvironmentWithoutStacks.serializableDataMap = serializableDataMapCopy;
+
+        runEnvironmentWithoutStacks.returnValues =
+                (ReturnValues) SerializationUtils.clone(returnValues);
+
+        runEnvironmentWithoutStacks.nextStepPosition = nextStepPosition;
+
+        return runEnvironmentWithoutStacks;
```

Comment:
in that case we should add setter methods in RunEnv for the ContextStack and ParentFlowStack  fields / add `reset` methods to the ..Stack classes

`.. @orius123 WDYT?`